pkg/logproto/logproto.proto

```diff
@@ -452,7 +452,7 @@ message DetectedFieldsRequest {
 
 message DetectedFieldsResponse {
   repeated DetectedField fields = 1;
-  uint32 fieldLimit = 2;
+  uint32 fieldLimit = 2 [(gogoproto.jsontag) = "fieldLimit,omitempty"];
 }
```

Comment:
ideally yes, but I don't know how to do that without having to implement a new method on both the querier and query frontend, which also feels messy since the query frontend shouldn't be concerned with returning internal representations of anything.

I think I could probably learn from the quantile/probabilistic query types code you did. I can circle back on this once I've had a chance to get more familiar with that?